app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/CachedFirestoreDatabaseManagement.kt

```diff
@@ -0,0 +1,69 @@
+package com.github.HumanLearning2021.HumanLearningApp.firestore
+
+import android.content.Context
+import android.net.Uri
+import android.util.Log
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.Converters
+import com.github.HumanLearning2021.HumanLearningApp.offline.CachePictureRepository
+import com.github.HumanLearning2021.HumanLearningApp.offline.OfflineCategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.offline.PictureRepository
+
+class CachedFirestoreDatabaseManagement internal constructor(
+    val dbName: String
+): FirestoreDatabaseManagement(FirestoreDatabaseService(dbName)) {
```

Comment:
But then I would not be able to override the needed methods? Also we won't have another cloud service. Caching is useless for the OfflineDatabaseManagement because it already stores images locally.

Reply:
ok